ykrt/src/mt.rs

```diff
@@ -318,6 +210,9 @@ impl MT {
                             // We've initialised this Location and obtained the lock, so we can now
                             // start tracing for real.
                             let tid = Arc::clone(&mtt.tid);
+                            // FIXME: this and other jit-state prints should be behind an opt-in
+                            // `jit-state-print` feature.
+                            eprintln!("jit-state: start-tracing");
```

Comment:
Oh, god, my poor code polluted by this monstrosity! 📦:p

[In even the medium term, this is actually really hard to do properly: what does it mean in a multi-threaded context? I doubt that `print`s are the right way of doing it. I'll fight that battle another day!]

Reply:
To be clear, these will be guarded so that they are only present for testing.

This is the same semantics as what we do currently (albeit it used to be in generated IR in ykllvm). I had to introduce them to have the current tests pass.

(We should probably prefix the events with some kind of thread identifier though).

Reply:
I've realised that this sort of testing really isn't going to work well in multi-threading. I understand that we need to do *something* but it's not going to be simple print statements. Maybe global counters will be the way to go -- I need to think a bit more.

Reply:
I've feature-gated those for now. See 7a4e8799dc9.

Reply:
Good idea!